department-of-veterans-affairs/caseflow
# Matt&Billy/APPEALS-10172

Resolves [APPEALS-10172](https://vajira.max.gov/browse/APPEALS-10172)

### Description
Billy noticed a bug where, whenever cancelling an intake while on the "Add Claimant" page, it would not redirect you back to the Intake homepage. This could cause users to seemingly become trapped (though, workarounds were confirmed to exist). Continuing to proceed with form population after the failed redirect could cause further errors, but no noticeable harm.

Edit: This issue has been observed on the AddPoaPage as well.

### Acceptance Criteria
- [ ] Users are redirected from the "Add Claimant" page to the form selection page of intake whenever they cancel an intake while on the "Add Claimant" form.
- [ ] Users are redirected from the "Add Claimant's POA" page to the form selection page of intake whenever they cancel an intake while on the "Add Claimant's POA" form.

### Testing Plan
1. Log into Caseflow as an Intake user (ex: `BVADWISE`)
2. Navigate to the Intake homepage ([Local link](http://localhost:3000/intake/))
3. Begin a 10182 intake.
4. On the Review page, select "Yes" for the prompt "Is the claimant someone other than the Veteran?".
5. Select "Claimant not listed" in the radio fieldset that will appear as a result.
![localhost_3000_intake_add_claimant (6)](https://user-images.githubusercontent.com/99351305/195235812-1f837050-4496-4847-ae7d-15dd3e631e75.png)

6. Once the required fields are populated, click the "Continue to next step" button.
7. You will now be on the "Add Claimant" page.
![localhost_3000_intake_add_claimant (7)](https://user-images.githubusercontent.com/99351305/195235831-acbbb7f4-f792-4650-94f9-d54a3f9cb432.png)

8. Click on "Cancel intake". Select any of the cancellation reasons, and then click the submission button.
![localhost_3000_intake_add_claimant (8)](https://user-images.githubusercontent.com/99351305/195235856-bad1c1aa-5320-44ee-91e4-78d91af4a033.png)

9. Ensure you are redirected back to the form selection page (`/intake/`).

10. Repeat steps 1-7.
11. This time, select a relationship from the "Relationship to the Veteran" dropdown.
12. Fill in all required fields. Select "Yes" for the prompt "Do you have a VA Form 21-22 for this cl
aimant?"
![localhost_3000_intake_add_claimant (9)](https://user-images.githubusercontent.com/99351305/195354705-4914baa8-07f3-426e-a334-e55d71abb49a.png)

13. You will now be on the "Add Claimant's POA" page (`AddPoaPage.jsx`)
![localhost_3000_intake_add_claimant (11)](https://user-images.githubusercontent.com/99351305/195355029-067444d7-5d72-4065-8902-a6ba2750f2d9.png)

14. Click on "Cancel intake". Like before, select any of the options on the modal and then click on the submission button.
15. Ensure you are redirected back to the form selection page (`/intake/`).
